test/lib/connection/connector/postmessage-legacy.test.js

```diff
@@ -70,38 +70,26 @@ describe('connector/postmessage-legacy', function() {
           stubPostMessage(info);
         });
 
-        it(command.ios, async () => {
+        it('should reject on unknown commands', async () => {
           let sendFn = await connect();
-          let ios = await sendFn(command.ios);
-          expect(ios).to.equal(true);
+          return expect(sendFn('unknown-asdf-command')).to.be.rejected;
         });
 
-        it(command.android, async () => {
-          let sendFn = await connect();
-          let android = await sendFn(command.android);
-          expect(android).to.equal(false);
-        });
-
-        it(command.mobile, async () => {
-          let sendFn = await connect();
-          let mobile = await sendFn(command.mobile);
-          expect(mobile).to.equal(true);
-        });
-
-        it(command.version, async () => {
-          let sendFn = await connect();
-          let version = await sendFn(command.version);
-          expect(version).to.equal(mockVersion);
-        });
-
-        it('throw error on unknown commands', async () => {
-          let sendFn = await connect();
-          let data = 'Command unknown not supported by driver';
-
-          try {
-            await sendFn('unknown');
-          } catch (error) {
-            expect(error.message).to.equal(data);
+        describe('accepts all comands', async function() {
+          // mock window open
+          window.open = function() {};
+
+          let commandData = {
+            prefContentLang: ['de_DE', 'en_US']
+          };
+
+          for (let cmd in command) {
+            if (command.hasOwnProperty(cmd)) {
+              it('command.' + cmd, async () => {
+                let sendFn = await connect();
+                return expect(sendFn(command[cmd], commandData[cmd])).to.be.fulfilled;
+              });
+            }
```

Comment:
I like this generic test but are you still able to easily see affected command of a failure?

Reply:
Yes you are

Reply:
The problem here is more, that we don't test if the returned value is correct. Only that the promise is fullfilled. But then we had to mock the results generically too.